### PR TITLE
fix(web): fixes content-editable error that occurred on focus changes 🏠

### LINKED ISSUE #10620: bug(web): `Uncaught TypeError: this.getTextBeforeCaret() is undefined`
Reported by @jmdurdin. KMW 16.0.145.

```
Uncaught TypeError: this.getTextBeforeCaret() is undefined

    getDeadkeyCaret contentEditable.ts:105
    buildTransformFrom outputTarget.ts:122
    buildTranscriptionFrom outputTarget.ts:200
    process kbdInterface.ts:1081
    processNewContextEvent kbdInterface.ts:1014
    processNewContextEvent keyboardProcessor.ts:239
    processNewContextEvent inputProcessor.ts:73
    resetContext inputProcessor.ts:358
    resetContext kmwbase.ts:509
    _ControlBlur domEventHandlers.ts:200
    attachDOMEvent kmwutils.ts:199
    enableInputElement domManager.ts:342
    attachToControl domManager.ts:442
    _SetupDocument domManager.ts:713
    init domManager.ts:1547
    init kmwbase.ts:490
    init t.html:7
    <anonymous> t.html:41
    EventListener.handleEvent* t.html:41

contentEditable.ts:105:13
```

Sample page which produced the error:

```
<html>
<head>
  <script src='https://s.keyman.com/kmw/engine/16.0.145/keymanweb.js'></script>
  <script src='https://s.keyman.com/kmw/engine/16.0.145/kmwuibutton.js'></script>
  <script>
    init = function () {
      keyman.init({
        attachType: 'auto'
      }).then(function () {
        keyman.addKeyboards('@th', '@lo');
        var pt = document.getElementById('thai'),
          pl = document.getElementById('lao');
        ps = document.getElementById('cd');
        if (pt) keyman.setKeyboardForControl(pt, 'Keyboard_basic_kbdth0');
        if (pl) keyman.setKeyboardForControl(pl, 'Keyboard_basic_kbdlao');
        if (ps) keyman.setKeyboardForControl(ps, 'Keyboard_basic_kbdth0');
      });
    }

    window.addEventListener('load', () => { init(); });
  </script>
  <style>
    .main {
      padding: 4em;
    }
    input {
      margin: 2em;
      width: 10em
    }
    div#cd {
      display: block;
      position: relative;
      left: 4em;
      top: 0;
      height: 4em;
      width: 40em;
      border: 1px solid blue;
      background-color: pink;
    }
  </style>
</head>
<body>
  <div class='main'>
    <div contenteditable id='cd'></div>
    <label for 'thai'>Thai</label>
    <input id='thai' />
    <label for 'lao'>Lao</label>
    <input id='lao' />
  </div>
</body>
</html>
```

### LINKED ISSUE #10693: bug(web): `Uncaught TypeError: this.getTextBeforeCaret() is undefined`
There is still a consistent error when moving the focus from an attached content-editable element to elsewhere on the page (other than another attached element):
 
```
Uncaught TypeError: this.getTextBeforeCaret() is undefined
    getDeadkeyCaret contentEditable.ts:105
    buildTransformFrom outputTarget.ts:122
    buildTranscriptionFrom outputTarget.ts:200
    process kbdInterface.ts:1081
    processNewContextEvent kbdInterface.ts:1014
    processNewContextEvent keyboardProcessor.ts:239
    processNewContextEvent inputProcessor.ts:73
    resetContext inputProcessor.ts:358
    resetContext kmwbase.ts:509
    _ControlBlur domEventHandlers.ts:200
contentEditable.ts:105:13
```

Sample page:
```
<html>
	<head>
		<meta http-equiv="Content-Type" content="text/html; charset=UTF-8" />
		<meta name="viewport" content="width=device-width, user-scalable=no" />
		<meta name="format-detection" content="telephone=no" />
		<meta name="apple-mobile-web-app-status-bar-style" content="black-translucent" />
		<meta http-equiv="X-UA-Compatible" content="IE=edge" >
		<title>KeymanWeb Testing </title>
		<script src='https://s.keyman.com/kmw/engine/16.0.145/keymanweb.js'></script>
		<script src='https://s.keyman.com/kmw/engine/16.0.145/kmwuibutton.js'></script>
		<link rel='stylesheet' href='/kmw/kmwosk.css' type='text/css; charset=utf-8' />
 
		<script>
			init = function() {
				keyman.init({
					attachType:'manual'
					}).then(function() {
						keyman.addKeyboards('@th','@lo','@en');			
						var pt = document.getElementById('thai'),
								pl = document.getElementById('lao');
								ps = document.getElementById('cd');
						if(pt) keyman.attachToControl(pt);
						if(pl) keyman.attachToControl(pl);
						if(ps) keyman.attachToControl(ps);
						if(pt) keyman.setKeyboardForControl(pt,'Keyboard_basic_kbdth0');
						if(pl) keyman.setKeyboardForControl(pl,'Keyboard_basic_kbdlao');
						if(ps) keyman.setKeyboardForControl(ps,'Keyboard_basic_kbdth0');
					});
			}			
			window.addEventListener('load',()=>{init();});
		</script>
		<style>
			.main{padding:4em;}
			input {margin:2em; width:10em}
			div#cd {display:block;position:relative; left:4em; top:0; height:4em;width:40em;border:1px solid blue;background-color:pink;}

		</style>
	</head>
	<body>
		<div class='main'>
		<p> Click on the content editable (pink) area then to a blank area of the window </p>
		<div contenteditable id='cd'></div>
			<label for 'thai'>Thai</label>
			<input id='thai' />
			<label for 'lao'>Lao</label>
			<input id='lao' />
		</div>
	</body>
</html>
```

Reported by @jmdurdin

### PR DESCRIPTION
**Pull request by jahorton**
Fixes #10620.
Fixes #10693.

@keymanapp-test-bot skip

### CODE REVIEW

File: web/source/dom/targets/contentEditable.ts [hunk 1]
```diff
@@ -107,7 +107,7 @@ namespace com.keyman.dom.targets {
 
     getTextBeforeCaret(): string {
       if(!this.hasSelection()) {
-        return;
+        return this.getText();
```
Comment (jahorton) [inline web/source/dom/targets/contentEditable.ts:110]: This matches the fix seen here, which is already part of 17.0:

https://github.com/keymanapp/keyman/blob/66142be3930c24f996939f7496a377f23223e3c3/web/src/engine/element-wrappers/src/contentEditable.ts#L109-L112

**Comment by mcdurdin** [comment]
Does this relate to a specific reported issue?

**Review by mcdurdin — APPROVED** [review]
LGTM but would like to link to a reported issue before we merge into 16.0.

**Comment by jahorton** [comment]
The .getTextBeforeCaret issues.  There should be two.  (Will link when not on phone.)

**Comment by keyman-server** [comment]
Changes in this pull request will be available for download in [Keyman version 16.0.146](https://keyman.com/downloads/releases/stable/16.0.146)

**Pull request merged**